nextstrain/cli/command/check_setup.py

```diff
@@ -111,17 +115,26 @@ def run(opts: Options) -> int:
             config.set("core", "runner", runner_name(default_runner))
             default_runner.set_default_config()
     else:
-        print(failure("No support for any Nextstrain environment."))
+        if set(opts.runners) == set(all_runners):
+            print(failure("No support for any Nextstrain environment."))
+        else:
+            print(failure("No support for selected Nextstrain environments."))
 
     print()
     if default_runner in supported_runners:
         print(f"All good!  Default environment ({runner_name(default_runner)}) is supported.")
     else:
-        print(failure(f"No good.  Default environment ({runner_name(default_runner)}) is not supported."))
+        if default_runner in opts.runners:
+            print(failure(f"No good.  Default environment ({runner_name(default_runner)}) is not supported."))
+        else:
+            print(f"Warning: Support for the default environment ({runner_name(default_runner)}) was not checked.")
 
         if supported_runners and not opts.set_default:
             print()
             print("Try running check-setup again with the --set-default option to pick a supported runner above.")
 
     # Return a 1 or 0 exit code
-    return int(0 if default_runner in supported_runners else 1)
+    if default_runner in opts.runners:
+        return 0 if default_runner in supported_runners else 1
+    else:
+        return 0 if supported_runners else 1
```

Comment:
Ah, this is intentional. I think "what exit codes are returned when" is much clearer with the code as-is rather than using a mutable variable throughout the rest of the logic. Keeping the split also means that as the messaging logic changes, the exit logic doesn't also have to change.